buildSrc/subprojects/cleanup/src/main/kotlin/gradlebuild/cleanup.gradle.kts

```
@@ -16,49 +16,20 @@
 package gradlebuild
 
 import gradlebuild.basics.BuildEnvironment
-import gradlebuild.cleanup.extension.CleanupExtension
-import gradlebuild.cleanup.services.DaemonTracker
-import gradlebuild.cleanup.tasks.CleanUpCaches
-import gradlebuild.cleanup.tasks.CleanUpDaemons
-import gradlebuild.cleanup.tasks.KillLeakingJavaProcesses
 
 plugins {
     base
-    id("gradlebuild.module-identity") // Would be good to avoid this in the root project, and instead apply a 'clean' Plugin to each subproject
 }
 
-tasks.register<CleanUpCaches>("cleanUpCaches") {
-    version.set(moduleIdentity.version)
-    homeDir.set(layout.projectDirectory.dir("intTestHomeDir"))
-}
-val trackerService = gradle.sharedServices.registerIfAbsent("daemonTracker", DaemonTracker::class) {
-    parameters.gradleHomeDir.fileValue(gradle.gradleHomeDir)
-    parameters.rootProjectDir.set(rootProject.layout.projectDirectory)
-}
-extensions.create<CleanupExtension>("cleanup", trackerService)
-tasks.register<CleanUpDaemons>("cleanUpDaemons") {
-    this.tracker.set(trackerService)
-}
-
-val killExistingProcessesStartedByGradle by tasks.registering(KillLeakingJavaProcesses::class) {
-    tracker.set(trackerService)
-}
-
-// TODO find another solution here to avoid reaching into another project. Maybe the CI job should do 'killExistingProcessesStartedByGradle' first directly.
-if (BuildEnvironment.isCiServer) {
-    tasks {
-        val cleanTask = named("clean") {
-            dependsOn(killExistingProcessesStartedByGradle)
-        }
-        subprojects {
-            this.tasks.configureEach {
-                mustRunAfter(killExistingProcessesStartedByGradle)
+if (BuildEnvironment.isCiServer && rootProject.name != "test") {
+    val killExistingProcessesStartedByGradle by rootProject.tasks.getting
+    val rootCleanTask = rootProject.tasks.named("clean")
```

:x: Since we apply the 'base' plugin and the accessors are lazy in Kotlin DSL, it can be used here:

```suggestion
    val rootCleanTask = rootProject.tasks.clean
```